src/credentials/CertificationDeclaration.cpp

```diff
@@ -236,6 +265,15 @@ CHIP_ERROR DecodeCertificationElements(const ByteSpan & encodedCertElements, Cer
     }
     VerifyOrReturnError(err == CHIP_END_OF_TLV || err == CHIP_ERROR_UNEXPECTED_TLV_ELEMENT || err == CHIP_NO_ERROR, err);
 
+    if (reader.GetTag() == ContextTag(kTag_AuthorizedPAAList) && reader.GetType() == kTLVType_Array)
+    {
+        certDeclContent.authorizedPAAListPresent = true;
+
+        // skip optional Authorized PAA List
+    }
+    err = reader.Next();
```

Comment:
Again, feels like this should be inside the block.

Also, shouldn't we be verifying that the PAA list has the right structure, if it's present, even though we are not doing anything with it?  But I'd think we should add it to `CertificationElementsWithoutPIDs`...

Reply:
fixed. Also added code to verify that the structure of this list is correct. 

Following this change I also added similar verification logic to the ProductIds list, which is also skipped